### PR TITLE
fix: update outdated links to documentation

### PR DESCRIPTION
**Pull request by krauspt**
I've found several outdated links to Starknet documentation and one link to Nethermind docs. 

This PR updates all links to actual documentation while preserving the linked topic.

### CODE REVIEW

**Comment by krauspt** [comment]
@omerfirmak @wojciechos is there anything from my side that needs to be done?

**Comment by krauspt** [comment]
@omerfirmak @wojciechos is there anything from my side that needs to be done?

**Comment by IronGauntlets** [comment]
In hindsight, I think it was not a good idea to add those comments as keeping them up to date is extra overhead. 

It would be better to just remove them. if you want to do it please go ahead. There are similar comments in `rpc/handlers.go`. Otherwise, we can close this PR and revisit this at some later point.

**Review by IronGauntlets — APPROVED** [review]

File: core/trie/trie.go [hunk 1]
```diff
@@ -118,7 +118,7 @@ func isSubset(longerKey, shorterKey *Key) bool {
 // path is suffix of key that diverges from parentKey. For example,
 // for a key 0b1011 and parentKey 0b10, this function would return the path object of 0b0.
 //
-// [specification]: https://docs.starknet.io/documentation/develop/State/starknet-state/
+// [specification]: https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/starknet-state/
```
Comment (derrix060) [inline core/trie/trie.go:121]:
```suggestion
// [specification]: https://docs.starknet.io/architecture-and-concepts/network-architecture/starknet-state/
```

File: core/trie/trie.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ type hashFunc func(*felt.Felt, *felt.Felt) *felt.Felt
 //   - key: represents the storage key for trie [Node]s. It is the full path to the node from the
 //     root.
 //
-// [specification]: https://docs.starknet.io/documentation/develop/State/starknet-state/
+// [specification]: https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/starknet-state/
```
Comment (derrix060) [inline core/trie/trie.go:34]:
```suggestion
// [specification]: https://docs.starknet.io/architecture-and-concepts/network-architecture/starknet-state/
```

File: core/trie/node.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func (n *Node) Hash(path *Key, hashFunc hashFunc) *felt.Felt {
 	}
 
 	pathFelt := path.Felt()
-	// https://docs.starknet.io/documentation/develop/State/starknet-state/
+	// https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/starknet-state/
```
Comment (derrix060) [inline core/trie/node.go:27]:
```suggestion
	// https://docs.starknet.io/architecture-and-concepts/network-architecture/starknet-state/
```

File: core/crypto/pedersen_hash.go [hunk 4]
```diff
@@ -8,15 +8,15 @@ import (
...
+// [Pedersen array hashing]: https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#array_hashing
 func PedersenArray(elems ...*felt.Felt) *felt.Felt {
 	var digest PedersenDigest
 	return digest.Update(elems...).Finish()
 }
 
 // Pedersen implements the [Pedersen hash].
 //
-// [Pedersen hash]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#pedersen_hash
+// [Pedersen hash]: https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#pedersen_array_hash
```
Comment (derrix060) [inline core/crypto/pedersen_hash.go:19]:
```suggestion
// [Pedersen hash]: https://docs.starknet.io/architecture-and-concepts/cryptography/hash-functions/#pedersen_array_hash
```

File: core/crypto/pedersen_hash.go [hunk 5]
```diff
@@ -8,15 +8,15 @@ import (
 
 // PedersenArray implements [Pedersen array hashing].
 //
-// [Pedersen array hashing]: https://docs.starknet.io/documentation/develop/Hashing/hash-functions/#array_hashing
+// [Pedersen array hashing]: https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#array_hashing
```
Comment (derrix060) [inline core/crypto/pedersen_hash.go:11]:
```suggestion
// [Pedersen array hashing]: https://docs.starknet.io/architecture-and-concepts/cryptography/hash-functions/#array_hashing
```

File: core/crypto/keccak.go [hunk 6]
```diff
@@ -7,7 +7,7 @@ import (
 
 // StarknetKeccak implements [Starknet keccak]
 //
-// [Starknet keccak]: https://docs.starknet.//io/documentation/develop/Hashing/hash-functions/#starknet_keccak
+// [Starknet keccak]: https://docs.starknet.io/documentation/architecture_and_concepts/Cryptography/hash-functions/#starknet_keccak
```
Comment (derrix060) [inline core/crypto/keccak.go:10]:
```suggestion
// [Starknet keccak]: https://docs.starknet.io/architecture-and-concepts/cryptography/hash-functions/#starknet_keccak
```

File: core/contract.go [hunk 7]
```diff
@@ -68,7 +68,7 @@ func ContractAddress(callerAddress, classHash, salt *felt.Felt, constructorCallD
 	prefix := new(felt.Felt).SetBytes([]byte("STARKNET_CONTRACT_ADDRESS"))
 	callDataHash := crypto.PedersenArray(constructorCallData...)
 
-	// https://docs.starknet.io/documentation/architecture_and_concepts/Contracts/contract-address
+	// https://docs.starknet.io/documentation/architecture_and_concepts/Smart_Contracts/contract-address/
```
Comment (derrix060) [inline core/contract.go:71]:
```suggestion
	// https://docs.starknet.io/architecture-and-concepts/smart-contracts/contract-address/
```

File: core/crypto/pedersen_hash.go (hunk 4 above)
Comment (kirugan) [reply core/crypto/pedersen_hash.go:19]: I think this one should point to `#pedersen_hash` anchor

**Comment by derrix060** [comment]
Closed in favor or #1929

**Pull request closed without merging**